prefetch.bs

```diff
@@ -713,6 +714,18 @@ This section contains patches to [[NAVIGATION-TIMING]].
     Add an additional parameter to [=create the navigation timing entry=], which is a [=string=] <var ignore>deliveryType</var>, and pass it as an additional argument to [=setup the resource timing entry=].
 </div>
 
+<h2 id="interaction-with-other-specs">Interaction with other specifications and concepts</h2>
+
+<div algorithm="clear prefetch cache for origin patch">
+  Modify [$clear cache for origin$] to add the following step at the end.
+
+  1. [=list/For each=] [=navigable/top-level traversable=] |traversable| in the user agent's [=top-level traversable set=]:
+    1. Let |navigables| be the [=list=] of [=Document/inclusive descendant navigables=] from |traversable|'s [=navigable/active document=].
+    2. Let |documents| be the [=list=] of |navigables|' [=navigable/active document=]s.
+    3. [=list/For each=] [=navigable/active document=] |activeDocument| in |documents|:
+      1. [=prefetch record/cancel and discard=] |activeDocument|'s [=Document/prefetch records=].
```

Comment:
You need to loop over each prefetch record. "Cancel and discard" takes a prefetch record + a document, but `|activeDocument|'s [=Document/prefetch records=]` is a list. Make sure to match up the type signatures at the call site and declaration site.

Reply:
Yeah I agree, I was trying to prevent having too many loops, but this way is more readable. Realizing the spec is written very much how code would function, but just in spec related syntax.